testing: Fix pod delete timeout failures after InPlacePodVerticalScaling Graduate to Beta commit

esotsal opened the pull request:
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

Please check analysis [here](https://github.com/kubernetes/kubernetes/issues/128837#issuecomment-2488086590) and conversation in this PR below.

In brief this commit modifies testing code, and ensure same delete timeout duration is used, more specificaly 

- replaced DeletePod timeouts used before with framework.timeouts.PodDelete, previous timeouts where less than 5 minutes, please see full commit for details. 
  https://github.com/kubernetes/kubernetes/blob/35d098aaa0a912171c46345e2c0a7b54d6046ef4/test/e2e/framework/timeouts.go#L26

- Replace of Delete with DeleteSync pod_resize test, with a 5 minute timeout, to ensure Pods are Deleted and will not impact some tests which were not resilient with the existence of pods.
 
#### Which issues this PR fixes

Relates to #128837
Fixes test failures due to pods not deleted with the introduction of InPlacePodVerticalScaling Beta 

#### Special notes for your reviewer:


Please see files changes for more details.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

Commit:
Fix pod delete issues in podresize tests

esotsal commented:
/cc @tallclair @hshiina @AnishShah

esotsal commented:
/triage accepted

kannon92 commented:
/test pull-kubernetes-e2e-capz-windows-master

kannon92 commented:
/test pull-kubernetes-node-kubelet-serial-containerd

/hold

Let’s confirm that this fixes the ci issues before merge.

dims commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers

aojea commented on test/e2e/common/node/pod_resize.go:
```diff
@@ -1006,7 +1008,7 @@ func doPodResizeTests(f *framework.Framework) {
 
 			ginkgo.By("deleting pod")
-			framework.ExpectNoError(podClient.Delete(ctx, newPod.Name, metav1.DeleteOptions{}))
+			podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, timeouts.PodDelete)
```
why is the Delete call erroring ?
why should we ignore the error?

ffromani replied:
I agree using `DeleteSync` is better and very likely to fix or at least significantly improve flakiness/failures we observed lately.
But I also agree we should not ignore the error or at very least clearly documenting why.

aojea replied:
my bad, pod client.DeleteSync checks for the error internally

https://github.com/kubernetes/kubernetes/blob/c9092f69fc0c099062dd23cd6ee226bcd52ec790/test/e2e/framework/pod/pod_client.go#L219-L225

/lgtm

ffromani replied:
got it thanks for the extra explanation @esotsal

tallclair replied:
These tests used to use DeleteSync, but it significantly slows down the tests, since there are so many cases.

Are the flakes due to leaving so many pods in a pending deletion state?

/cc @AnishShah

esotsal replied:
@ffromani that's true, but to be fair i think there is room for improvement, for example  Before and After each test to check system under test meets expectations i.e. there are no other pods running or use a labeling mechanism ( as demonstrated [here](https://github.com/kubernetes/kubernetes/issues/128837#issuecomment-2489315057) )

 @tallclair i think stability should be more important to test duration

hshiina replied:
Could we replace the command in a pod for resize tests with `InfiniteSleepCommand` or insert `trap` for handling SIGTERM? I think this blocks pod deletion for 30 seconds:
https://github.com/kubernetes/kubernetes/blob/c9092f69fc0c099062dd23cd6ee226bcd52ec790/test/e2e/framework/pod/resize.go#L161

ffromani replied:
If the pods terminates faster that would help a lot, and we can and should apply the same fix in quite other places, but still we will need to have guaranteed cleanup between tests in the lane marked `serial` (or to rethink from scratch a bunch of resource management tests)

pacoxu replied:
> @tallclair i think stability should be more important to test duration. Yes the flakes are due to undeleted pods, please check [here](https://github.com/kubernetes/kubernetes/issues/128837#issuecomment-2489315057)

Does this mean that we do not have to revert the promotion https://github.com/kubernetes/kubernetes/pull/128875?

tallclair replied:
We still need to fix https://github.com/kubernetes/kubernetes/issues/128917 (although it's an easy fix), and I don't think the windows test failures are due to undeleted pods (though I could be wrong).

esotsal replied:
> Could we replace the command in a pod for resize tests with `InfiniteSleepCommand` or insert `trap` for handling SIGTERM? I think this blocks pod deletion for 30 seconds:
> 
> https://github.com/kubernetes/kubernetes/blob/c9092f69fc0c099062dd23cd6ee226bcd52ec790/test/e2e/framework/pod/resize.go#L161

Missed this one , sorry @hshiina . acknowledged, will give it a try as well.

aojea commented:
hmm,  pull-kubernetes-e2e-gce	https://github.com/kubernetes/kubernetes/commit/55ad72d58f353c4060981af4008ee06acac1c9f4

```

Kubernetes e2e suite: [It] [sig-node] Pod InPlace Resize Container pod-resize-resource-quota-test expand_less	24s
{ failed [FAILED] exceeded quota: resize-resource-quota, requested: memory=350Mi, used: memory=700Mi, limited: memory=800Mi
Expected an error to have occurred.  Got:
    <nil>: nil
In [It] at: k8s.io/kubernetes/test/e2e/node/pod_resize.go:172 @ 11/20/24 12:31:22.502
}
```

ffromani commented:
/priority critical-urgent

kannon92 commented:
> [pull-kubernetes-e2e-capz-windows-master ](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-capz-windows-master/1859208062640852992) failures are not related with InPlacePodVerticalScaling , sig-storage / sig-windows

I disagree. https://github.com/kubernetes/kubernetes/pull/128875#issuecomment-2488223856

The revert of beta promotion seems to fix the windows test failures.

esotsal commented:
> The revert of beta promotion seems to fix the windows test failures.

Thanks @kannon92  for the explanation

esotsal commented:
> hmm, pull-kubernetes-e2e-gce [55ad72d](https://github.com/kubernetes/kubernetes/commit/55ad72d58f353c4060981af4008ee06acac1c9f4)
> 
> ```
> 
> Kubernetes e2e suite: [It] [sig-node] Pod InPlace Resize Container pod-resize-resource-quota-test expand_less	24s
> { failed [FAILED] exceeded quota: resize-resource-quota, requested: memory=350Mi, used: memory=700Mi, limited: memory=800Mi
> Expected an error to have occurred.  Got:
>     <nil>: nil
> In [It] at: k8s.io/kubernetes/test/e2e/node/pod_resize.go:172 @ 11/20/24 12:31:22.502
> }
> ```

Seems a flaky test it passed with second attempt [here](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-gce/1859226054917361664) , this return error is not related with this commit

ffromani commented:
> we have zero flake policy, we should fix those flaky tests or remove them, but having something that is flake generates uncertainty ... I'm happy to help to review and work on making the tests more solid if you want

Hi, unrelated question: where can I learn about the rationale and the decision process about the zero flakes policy?

esotsal commented:
> /test pull-kubernetes-node-kubelet-serial-containerd
> 
> /hold
> 
> Let’s confirm that this fixes the ci issues before merge.

[pull-kubernetes-node-kubelet-serial-containerd](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-node-kubelet-serial-containerd/1859223507045453825) passed @kannon92 , do you agree we remove the hold?

AnishShah commented on test/e2e/common/node/pod_resize.go:
```diff
@@ -932,6 +932,8 @@ func doPodResizeTests(f *framework.Framework) {
+	timeouts := f.Timeouts
+
 	for idx := range tests {
 		tc := tests[idx]
```
Another issue is that all subtests run in the same namespace. Running tests in separate namespaces might help too. I'm trying that in https://github.com/kubernetes/kubernetes/pull/128719/files#diff-925709f69f421bb42210b82f6e832d44fff45daedeee3accbc35ac1b5d4f49c3R1159-R1228

esotsal replied:
I think @AnishShah your proposal will improve things and will solve kubetest2 issue. I recall i had the same failures when i tried to migrate the e2e_node CPU Manager tests to e2e/node.

```
[FAILED] A suite timeout occurred and then the following failure was recorded in the timedout node before it exited:
Context was cancelled (cause: suite timeout occurred) after 67.125s.
There is no failure as the matcher passed to Consistently has not yet failed
In [It] at: k8s.io/kubernetes/test/e2e_node/node_container_manager_test.go:128 @ 11/20/24 13:57:28.46
}
```

kannon92 commented:
/hold cancel

esotsal commented:
@kannon92 created https://github.com/kubernetes/kubernetes/issues/128897#issue-2677073607  to dig into the pull-kubernetes-e2e-capz-windows-master

aojea commented:
> Hi, unrelated question: where can I learn about the rationale and the decision process about the zero flakes policy?

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md

To be honest, I don't expect someone try to justify having a flaky test is correct :)

aojea commented:
> Which test fails doesn’t matter , failure happens always on the last tes

why it fails? we need to find root causes

esotsal commented:
/retest

esotsal commented:
/test pull-kubernetes-e2e-capz-windows-master 

Testing again to gather more logs, i will keep this commit without ippvs revert intentionally to debug. pull-kubernetes-e2e-capz-windows-master passed checking https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=128880 passes without the revert https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/128880/pull-kubernetes-e2e-capz-windows-master/1859388054804893696

esotsal commented:
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers

esotsal commented:
> > Which test fails doesn’t matter , failure happens always on the last tes
> 
> why it fails? we need to find root causes

From kubetest2 last two failures run 

```
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat n1-standard-2-cos-stable-117-18613-75-37-b18ea0b6/n1-standard-2-cos-stable-117-18613-75-37-b18ea0b6-ginkgo.log | grep FAIL
  [FAILED] in [JustAfterEach] - k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:224 @ 11/21/24 04:30:47.143
  [FAILED] A suite timeout occurred and then the following failure was recorded in the timedout node before it exited:
FAIL! - Suite Timeout Elapsed -- 28 Passed | 1 Failed | 0 Pending | 693 Skipped
--- FAIL: TestE2eNode (1805.49s)
FAIL
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat n1-standard-2-cos-stable-117-18613-75-37-49312624/n1-standard-2-cos-stable-117-18613-75-37-49312624-ginkgo.log | grep FAIL
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/criproxy_test.go:110 @ 11/21/24 04:00:39.905
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/image_pull_test.go:156 @ 11/21/24 04:11:04.681
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/image_pull_test.go:73 @ 11/21/24 04:18:58.592
FAIL! - Suite Timeout Elapsed -- 21 Passed | 1 Failed | 0 Pending | 700 Skipped
--- FAIL: TestE2eNode (1834.63s)
FAIL
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat n1-standard-2-ubuntu-gke-2204-1-29-v20241115-13b2ca4f/n1-standard-2-ubuntu-gke-2204-1-29-v20241115-13b2ca4f-ginkgo.log | grep FAIL
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/container_restart_test.go:60 @ 11/21/24 04:05:46.621
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/image_pull_test.go:156 @ 11/21/24 04:12:50.828
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/container_restart_test.go:84 @ 11/21/24 04:24:19.243
  [FAILED] in [It] - k8s.io/kubernetes/test/e2e_node/container_lifecycle_test.go:1384 @ 11/21/24 04:30:42.519
  [FAILED] A suite timeout occurred and then the following failure was recorded in the timedout node before it exited:
FAIL! - Suite Timeout Elapsed -- 23 Passed | 1 Failed | 0 Pending | 698 Skipped
--- FAIL: TestE2eNode (1802.42s)
FAIL
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat kubetest2/n1-standard-2-
n1-standard-2-cos-stable-117-18613-75-37-cf2332a0/     n1-standard-2-cos-stable-117-18613-75-37-f69bafb4/     n1-standard-2-ubuntu-gke-2204-1-29-v20241115-6d35f80c/ 
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat kubetest2/n1-standard-2-cos-stable-117-18613-75-37-cf2332a0/n1-standard-2-cos-stable-117-18613-75-37-cf2332a0-ginkgo.log | grep FAIL
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/image_pull_test.go:156 @ 11/21/24 07:27:00.065
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/container_restart_test.go:60 @ 11/21/24 07:30:34.913
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/criproxy_test.go:58 @ 11/21/24 07:42:20.177
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:224 @ 11/21/24 07:47:58.333
  [FAILED] A suite timeout occurred and then the following failure was recorded in the timedout node before it exited:
FAIL! - Suite Timeout Elapsed -- 36 Passed | 1 Failed | 0 Pending | 685 Skipped
--- FAIL: TestE2eNode (1800.60s)
FAIL
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat kubetest2/n1-standard-2-
n1-standard-2-cos-stable-117-18613-75-37-cf2332a0/     n1-standard-2-cos-stable-117-18613-75-37-f69bafb4/     n1-standard-2-ubuntu-gke-2204-1-29-v20241115-6d35f80c/ 
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat kubetest2/n1-standard-2-cos-stable-117-18613-75-37-f69bafb4/n1-standard-2-cos-stable-117-18613-75-37-f69bafb4-ginkgo.log | grep FAIL
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e_node/criproxy_test.go:58 @ 11/21/24 07:35:24.462
FAIL! - Suite Timeout Elapsed -- 25 Passed | 1 Failed | 0 Pending | 696 Skipped
--- FAIL: TestE2eNode (1833.13s)
FAIL
sotiris@sotiris-Precision-5560:~/Documents/PR128880$ cat kubetest2/n1-standard-2-ubuntu-gke-2204-1-29-v20241115-6d35f80c/n1-standard-2-ubuntu-gke-2204-1-29-v20241115-6d35f80c-ginkgo.log | grep FAIL
  [FAILED] in [AfterEach] - k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:224 @ 11/21/24 07:48:01.195
  [FAILED] A suite timeout occurred and then the following failure was recorded in the timedout node before it exited:
FAIL! - Suite Timeout Elapsed -- 28 Passed | 1 Failed | 0 Pending | 693 Skipped
--- FAIL: TestE2eNode (1800.44s)
FAIL
```

I am confident that https://github.com/kubernetes/kubernetes/pull/128889 would improve things if combined with this commit. I will give it a try without rebasing to latest to give it a try.

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
> > Which test fails doesn’t matter , failure happens always on the last tes
 > 
 > 
 > why it fails? we need to find root causes
 
In addition to previous, from kubetest2 last failures discovered below remaining failure.

```
sotiris@sotiris-Precision-5560:~/Documents/PR128880/kubetest2_dyo$ cat n1-standard-2-ubuntu-gke-2204-1-29-v20241115-a3bc35f7/n1-standard-2-ubuntu-gke-2204-1-29-v20241115-a3bc35f7-ginkgo.log | grep -B2 FAIL
      k8s.io/kubernetes/test/e2e_node.init.func49.6.6.5({0x7fe391bdfd98, 0xc0017d3260})
      	k8s.io/kubernetes/test/e2e_node/memory_manager_test.go:558 +0xb5
  [FAILED] in [JustAfterEach] - k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:224 @ 11/21/24 10:13:31.937
--
      k8s.io/kubernetes/test/e2e_node.init.func49.5({0x7fe391bdfd98, 0xc001ce88d0})
      	k8s.io/kubernetes/test/e2e_node/memory_manager_test.go:353 +0xb5
  [FAILED] in [JustAfterEach] - k8s.io/kubernetes/test/e2e/framework/pod/pod_client.go:224 @ 11/21/24 10:14:01.951
--
```

Checking the code i noticed that DeleteSync was used with timeout 2*minutes.  

Performing a grep in the code repo discovered that not all tests use e2epod.DefaultPodDeletionTimeout ( which is 3 minutes ).  My theory is that with the updates of InPlacePodVerticalScaling BETA , the timeouts on those cases was not sufficient. To have a better control now ( but also in the future ) i have updated this commit and replaced timeouts below 3 minutes with e2epod.DefaultPodDeletionTimeout. I have also updated pod_resize_test and replaced framework timeout PodDelete ( which is equal 5 minutes ) to use e2epod.DefaultPodDeletionTimeout as well instead ( 3 minutes ).  Will run tests to test my theory.

```
sotiris@sotiris-Precision-5560:/usr/local/go/src/k8s.io/kubernetes$ git grep DeleteSync                                                                                                                               
pkg/controller/volume/persistentvolume/delete_test.go:func TestDeleteSync(t *testing.T) {                                                                                                                             
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, *metav1.NewDeleteOptions(15), e2epod.DefaultPodDeletionTimeout)                                                   
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, *metav1.NewDeleteOptions(15), e2epod.DefaultPodDeletionTimeout)                                                   
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, *metav1.NewDeleteOptions(2), e2epod.DefaultPodDeletionTimeout)           
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                         
test/e2e/common/node/lifecycle_hook.go:                 podClient.DeleteSync(ctx, podWithHook.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                         
test/e2e/common/node/pod_resize.go:                     podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, timeouts.PodDelete)                      
test/e2e/common/node/pod_resize.go:                     podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, timeouts.PodDelete)                                           
test/e2e/common/node/security_context.go:                               podClient.DeleteSync(ctx, createdPod1.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)
test/e2e/common/node/security_context.go:                               podClient.DeleteSync(ctx, createdPod2.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                
test/e2e/framework/pod/output/output.go:                podClient.DeleteSync(ctx, createdPod.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                 
test/e2e/framework/pod/pod_client.go:// DeleteSync deletes the pod and wait for the pod to disappear for `timeout`. If the pod doesn't                                                       
test/e2e/framework/pod/pod_client.go:func (c *PodClient) DeleteSync(ctx context.Context, name string, options metav1.DeleteOptions, timeout time.Duration) {                                   
test/e2e/network/conntrack.go:          e2epod.NewPodClient(fr).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                       
test/e2e/network/conntrack.go:          e2epod.NewPodClient(fr).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                               
test/e2e/network/conntrack.go:          e2epod.NewPodClient(fr).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                      
test/e2e/network/conntrack.go:          e2epod.NewPodClient(fr).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                                
test/e2e/network/loadbalancer.go:               e2epod.NewPodClient(f).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                        
test/e2e/network/loadbalancer.go:               e2epod.NewPodClient(f).DeleteSync(ctx, podBackend1, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                
test/e2e_node/container_log_rotation_test.go:                   ginkgo.DeferCleanup(e2epod.NewPodClient(f).DeleteSync, logRotationPod.Name, metav1.DeleteOptions{}, time.Minute)                 
test/e2e_node/container_log_rotation_test.go:                           ginkgo.DeferCleanup(e2epod.NewPodClient(f).DeleteSync, logRotationPod.Name, metav1.DeleteOptions{}, time.Minute)
test/e2e_node/cpu_manager_test.go:      e2epod.NewPodClient(f).DeleteSync(ctx, podName, delOpts, e2epod.DefaultPodDeletionTimeout)                                                    
test/e2e_node/critical_pod_test.go:                     e2epod.NewPodClient(f).DeleteSync(ctx, guaranteedPodName, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                           
test/e2e_node/critical_pod_test.go:                     e2epod.NewPodClient(f).DeleteSync(ctx, burstablePodName, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                            
test/e2e_node/critical_pod_test.go:                     e2epod.NewPodClient(f).DeleteSync(ctx, bestEffortPodName, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                           
test/e2e_node/critical_pod_test.go:                     e2epod.PodClientNS(f, kubeapi.NamespaceSystem).DeleteSync(ctx, criticalPodName, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)
test/e2e_node/device_manager_test.go:                   e2epod.NewPodClient(f).DeleteSync(ctx, testPod.Name, metav1.DeleteOptions{}, 2*time.Minute)                          
test/e2e_node/device_manager_test.go:                   e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, metav1.DeleteOptions{}, time.Minute)                                                             
test/e2e_node/device_manager_test.go:                           e2epod.NewPodClient(f).DeleteSync(ctx, p.Name, metav1.DeleteOptions{}, 2*time.Minute)                                                                 
test/e2e_node/device_plugin_test.go:                    e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, metav1.DeleteOptions{}, time.Minute)                       
test/e2e_node/device_plugin_test.go:                            e2epod.NewPodClient(f).DeleteSync(ctx, p.Name, metav1.DeleteOptions{}, 2*time.Minute)           
test/e2e_node/device_plugin_test.go:                    e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, metav1.DeleteOptions{}, time.Minute)                                                             
test/e2e_node/device_plugin_test.go:                    e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, deleteOptions, time.Minute)                                                                      
test/e2e_node/device_plugin_test.go:                    e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, deleteOptions, time.Minute)                                                                      
test/e2e_node/device_plugin_test.go:                    e2epod.NewPodClient(f).DeleteSync(ctx, devicePluginPod.Name, metav1.DeleteOptions{}, time.Minute)                                                             
test/e2e_node/device_plugin_test.go:                            e2epod.NewPodClient(f).DeleteSync(ctx, p.Name, metav1.DeleteOptions{}, 2*time.Minute)                                                                 
test/e2e_node/eviction_test.go:                         e2epod.NewPodClient(f).DeleteSync(ctx, spec.pod.Name, metav1.DeleteOptions{}, 10*time.Minute)                                          
test/e2e_node/garbage_collector_test.go:                                e2epod.NewPodClient(f).DeleteSync(ctx, pod.podName, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                     
test/e2e_node/hugepages_test.go:                        e2epod.NewPodClient(f).DeleteSync(ctx, testpod.Name, metav1.DeleteOptions{}, 2*time.Minute)                                         
test/e2e_node/image_gc_test.go:                 e2epod.NewPodClient(f).DeleteSync(ctx, pod.ObjectMeta.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                       
test/e2e_node/image_gc_test.go:                 e2epod.NewPodClient(f).DeleteSync(ctx, pod.ObjectMeta.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)                                                 
test/e2e_node/memory_manager_test.go:                   e2epod.NewPodClient(f).DeleteSync(ctx, testPod.Name, metav1.DeleteOptions{}, 2*time.Minute)                                             
test/e2e_node/memory_manager_test.go:                                   e2epod.NewPodClient(f).DeleteSync(ctx, testPod2.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)
test/e2e_node/memory_manager_test.go:                                           e2epod.NewPodClient(f).DeleteSync(ctx, workloadPod.Name, metav1.DeleteOptions{}, 2*time.Minute)         
test/e2e_node/node_perf_test.go:                e2epod.NewPodClient(f).DeleteSync(ctx, pod.Name, delOpts, e2epod.DefaultPodDeletionTimeout)                                                     
test/e2e_node/oomkiller_linux_test.go:                  e2epod.NewPodClient(f).DeleteSync(context.TODO(), testCase.podSpec.Name, metav1.DeleteOptions{}, framework.PodDeleteTimeout)
test/e2e_node/pod_hostnamefqdn_test.go:         ginkgo.DeferCleanup(e2epod.NewPodClient(f).DeleteSync, launchedPod.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)
test/e2e_node/podresources_test.go:                                     defer e2epod.NewPodClient(f).DeleteSync(ctx, pod.Name, metav1.DeleteOptions{}, time.Minute)       
test/e2e_node/resource_metrics_test.go:                 e2epod.NewPodClient(f).DeleteSync(ctx, pod0, metav1.DeleteOptions{GracePeriodSeconds: &zero}, 10*time.Minute)      
test/e2e_node/resource_metrics_test.go:                 e2epod.NewPodClient(f).DeleteSync(ctx, pod1, metav1.DeleteOptions{GracePeriodSeconds: &zero}, 10*time.Minute)                            
test/e2e_node/split_disk_test.go:                               e2epod.NewPodClient(f).DeleteSync(ctx, spec.pod.Name, metav1.DeleteOptions{}, 10*time.Minute)                 
test/e2e_node/util.go:                  e2epod.NewPodClient(f).DeleteSync(ctx, p.Name, metav1.DeleteOptions{}, 2*time.Minute)                                     

```

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

kannon92 commented:
Looking at pull-kubernetes-node-kubelet-serial-containerd-kubetest2, I think we are hitting a timeout due to default limits of ginkgo. We want this test to run for 4 hours and it seems to end around 40 minutes. This is unrelated to the issue at hand. 

I can fix this today.

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

pohly commented on test/e2e/common/node/pod_resize.go:
```diff
@@ -1006,7 +1006,7 @@ func doPodResizeTests(f *framework.Framework) {
 
 			ginkgo.By("deleting pod")
-			framework.ExpectNoError(podClient.Delete(ctx, newPod.Name, metav1.DeleteOptions{}))
+			podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, e2epod.DefaultPodDeletionTimeout)
```
Why is it okay to ignore errors?

esotsal replied:
We dont ignore the error , DeleteSync handles it internally. 
https://github.com/kubernetes/kubernetes/blob/c9092f69fc0c099062dd23cd6ee226bcd52ec790/test/e2e/framework/pod/pod_client.go#L217-L225

pohly commented on test/e2e_node/container_restart_test.go:
```diff
@@ -55,12 +55,11 @@ var _ = SIGDescribe("Container Restart", feature.CriProxy, framework.WithSerial(
-		})
-
 		ginkgo.It("Container restart backs off.", func(ctx context.Context) {
+			ginkgo.DeferCleanup(func() {
```
Please clean up: either call `ginkgo.DeferCleanup` after the code which causes the need to clean up (= in the `BeforeEach` above) or move that code here (= remove the `BeforeEach`).

esotsal replied:
Thanks, Acknowledged

esotsal replied:
Thanks, Done ( Note , i am copying the code from https://github.com/kubernetes/kubernetes/pull/128889/files here to perform tests before InPlacePodVerticalScaling Beta revert, if and when #128889 is merged, those will be removed during rebase )

tallclair reviewed (commented):
There are 3 different pod deletion timeouts defined:
1. `e2epod.PodDeleteTimeout` https://github.com/kubernetes/kubernetes/blob/40f222b6201d5c6476e5b20f57a4a7d8b2d71845/test/e2e/framework/pod/delete.go#L35
2. `e2epod.DefaultPodDeletionTimeout`
3. `framework.Timeouts.PodDelete`

I think the `f.Timeouts.PodDelete` is the right one to use here (and in the other places). We should probably just remove the first 2

esotsal commented:
> There are 3 different pod deletion timeouts defined:
> 
>     1. `e2epod.PodDeleteTimeout` https://github.com/kubernetes/kubernetes/blob/40f222b6201d5c6476e5b20f57a4a7d8b2d71845/test/e2e/framework/pod/delete.go#L35
> 
>     2. `e2epod.DefaultPodDeletionTimeout`
> 
>     3. `framework.Timeouts.PodDelete`
> 
> 
> I think the `f.Timeouts.PodDelete` is the right one to use here (and in the other places). We should probably just remove the first 2

Done

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
> I have [kubernetes/test-infra#33825](https://github.com/kubernetes/test-infra/pull/33825) up to fix the kubetest failures.

Thanks!

pacoxu commented:
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

carlory commented:
Before merging it, please take a look at https://github.com/kubernetes/kubernetes/issues/128897#issuecomment-2492941056. Thanks!

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
> Before merging it, please take a look at [#128897 (comment)](https://github.com/kubernetes/kubernetes/issues/128897#issuecomment-2492941056). Thanks!

Thanks changed description  to "relates to 128897"  ( unfortunately i still see github still has it as fix, if someone can guide me how to remove it from fix ? )

carlory commented:
> how to remove it from fix 

No worries. It's just a display issue. If it is merged, that issue won't be closed.

esotsal commented:
/test ci-kubernetes-e2e-windows-containerd-gce-master

esotsal commented:
> Before merging it, please take a look at [#128897 (comment)](https://github.com/kubernetes/kubernetes/issues/128897#issuecomment-2492941056). Thanks!

Thanks , update i am using this commit to test [this PR ](https://github.com/kubernetes/kubernetes/pull/128936) based on https://github.com/kubernetes/kubernetes/issues/128897#issuecomment-2493675700 with InPlacePodverticalScaling not been reverted. Credits to @hshiina for spotting this

dims commented on pkg/kubelet/kuberuntime/kuberuntime_manager.go:
```diff
@@ -549,6 +550,9 @@ func containerSucceeded(c *v1.Container, podStatus *kubecontainer.PodStatus) boo
 }
 
 func IsInPlacePodVerticalScalingAllowed(pod *v1.Pod) bool {
+	if runtime.GOOS == "windows" {
```
This looks darn important!!!

pohly replied:
And ideally, this should be a runtime check. Isn't there something to mark a test as "does not work on Windows"?

pacoxu replied:
We may move this function `IsInPlacePodVerticalScalingAllowed` to  `pkg/kubelet/kuberuntime/kuberuntime_manager_linux.go` and (`pkg/kubelet/kuberuntime/kuberuntime_manager_windows.go` or  `kuberuntime_manager_others.go`).

Add `"[LinuxOnly]"` to the e2e test?

esotsal replied:
Sounds worth trying, @pacoxu , @pohly @dims moving discussion [here](https://github.com/kubernetes/kubernetes/pull/128936#issuecomment-2505524573)

dims commented on test/e2e/common/node/pod_resize.go:
```diff
@@ -1006,7 +1006,7 @@ func doPodResizeTests(f *framework.Framework) {
 
 			ginkgo.By("deleting pod")
-			framework.ExpectNoError(podClient.Delete(ctx, newPod.Name, metav1.DeleteOptions{}))
+			podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
```
we are ok to skip checking the error here?

esotsal replied:
We dont ignore the error , DeleteSync handles it internally. 
https://github.com/kubernetes/kubernetes/blob/c9092f69fc0c099062dd23cd6ee226bcd52ec790/test/e2e/framework/pod/pod_client.go#L217-L225

dims replied:
Nice thanks @esotsal

dims commented on test/e2e/common/node/lifecycle_hook.go:
```diff
@@ -115,7 +115,7 @@ var _ = SIGDescribe("Container Lifecycle Hook", func() {
 			}
 			ginkgo.By("delete the pod with lifecycle hook")
-			podClient.DeleteSync(ctx, podWithHook.Name, *metav1.NewDeleteOptions(15), e2epod.DefaultPodDeletionTimeout)
+			podClient.DeleteSync(ctx, podWithHook.Name, *metav1.NewDeleteOptions(15), f.Timeouts.PodDelete)
```
Can you please call out what is the difference between the two timeouts and highlight it in the PR text?

dims replied:
I am guessing the new one is longer than the old one?

esotsal replied:
Yes, 

https://github.com/kubernetes/kubernetes/blob/35d098aaa0a912171c46345e2c0a7b54d6046ef4/test/e2e/framework/pod/pod_client.go#L47

https://github.com/kubernetes/kubernetes/blob/35d098aaa0a912171c46345e2c0a7b54d6046ef4/test/e2e/framework/timeouts.go#L26

dims replied:
please add this info to the PR text

esotsal replied:
Done

dims commented on test/e2e_node/criproxy_test.go:
```diff
@@ -91,26 +91,21 @@ var _ = SIGDescribe(feature.CriProxy, framework.WithSerial(), func() {
-		ginkgo.AfterEach(func() {
-			err := resetCRIProxyInjector(e2eCriProxy)
-			framework.ExpectNoError(err)
+			ginkgo.DeferCleanup(func() {
```
this pattern is way better. thanks!

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
I believe pull-kubernetes-e2e-gce-canary failures are not related with InPlacePodVerticalScaling, created https://github.com/kubernetes/kubernetes/issues/128942 to monitor the flaking.

esotsal commented:
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

hshiina commented:
It looks that currently no job runs tests with enabling `InPlacePodVerticalScaling`.

Especially, `pull-kubernetes-node-kubelet-serial-containerd-alpha-features` runs no test. Something may be wrong with its label filter:
https://github.com/kubernetes/test-infra/blob/c08c9c08ebb44b03557afeb5dd58b4e81bf9b914/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L660

esotsal commented:
> It looks that currently no job runs tests with enabling `InPlacePodVerticalScaling`.
> 
> Especially, `pull-kubernetes-node-kubelet-serial-containerd-alpha-features` runs no test. Something may be wrong with its label filter: https://github.com/kubernetes/test-infra/blob/c08c9c08ebb44b03557afeb5dd58b4e81bf9b914/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L660

For this commit , I have not reverted the beta to debug the tests. So it might be why the  alpha to not run the tests?  the non-alpha specific should run the tests.

esotsal commented:
I think I will rebase this commit, findings are sufficient. Thank you all for the comments and feedback.

esotsal commented:
> > It looks that currently no job runs tests with enabling `InPlacePodVerticalScaling`.
> > Especially, `pull-kubernetes-node-kubelet-serial-containerd-alpha-features` runs no test. Something may be wrong with its label filter: https://github.com/kubernetes/test-infra/blob/c08c9c08ebb44b03557afeb5dd58b4e81bf9b914/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L660
> 
> Thanks for letting us know, for this commit , I have not reverted the beta to debug the issue, InPlacePodVerticalScaling should be enabled by default. Being beta , might explain why pull-kubernetes-node-kubelet-serial-containerd-alpha-features does not run the tests?

Will add it as an extra step when the beta I merged  back not to forget

esotsal commented:
/test pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2
/test pull-kubernetes-node-kubelet-serial-podresize
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-e2e-gce-cos-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
/test pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2
/test pull-kubernetes-node-kubelet-serial-podresize
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-e2e-gce-cos-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd
/test pull-kubernetes-node-kubelet-serial-containerd-alpha-features
/test pull-kubernetes-node-kubelet-serial-containerd-sidecar-containers
/test pull-kubernetes-e2e-ec2
/test pull-kubernetes-e2e-gce-canary
/test pull-kubernetes-e2e-ec2-conformance
/test pull-kubernetes-e2e-capz-windows-master
/test pull-kubernetes-node-kubelet-serial-containerd-kubetest2

esotsal commented:
Will clean up this PR, keeping only the DeleteSync, removing https://github.com/kubernetes/kubernetes/pull/128889 and https://github.com/kubernetes/kubernetes/pull/128936 parts since they have their own PRs.  Will update the PR description and description.

esotsal commented:
/test pull-kubernetes-verify-lint

esotsal commented:
/hold cancel

esotsal commented:
/test all

esotsal commented:
/retest

pacoxu commented:
/test pull-kubernetes-e2e-gce-cos-alpha-features

esotsal commented:
Closing this PR moving to https://github.com/kubernetes/kubernetes/pull/129717 , needed because of company transfer. Will ping reviewers again in new PR.

pohly commented:
Why do you need to move your PRs? You are still using the same @esotsal GitHub handle.

There's theoretically a loophole where you had signed the CLA before and now no longer have, but then you also wouldn't be able to get your new PRs merged.

esotsal commented:
> Why do you need to move your PRs? You are still using the same @esotsal GitHub handle.
> 
> There's theoretically a loophole where you had signed the CLA before and now no longer have, but then you also wouldn't be able to get your new PRs merged.

I don't have access to Nordix repo any longer, so i need to move them under my esotsal.
